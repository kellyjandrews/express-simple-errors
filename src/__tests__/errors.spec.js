import { errors } from '../index';

describe('Error Classes', () => {
  it('allow Bad Request error', () => {
    const BadRequest = new errors.BadRequest();
    expect(BadRequest).toMatchSnapshot();
    expect(BadRequest.code).toBe(400);
    expect(BadRequest.message).toBe('Bad Request');
  });
  it('allow Unauthorized error', () => {
    const Unauthorized = new errors.Unauthorized();
    expect(Unauthorized).toMatchSnapshot();
    expect(Unauthorized.code).toBe(401);
    expect(Unauthorized.message).toBe('Unauthorized');
  });
  it('allow Forbidden error', () => {
    const Forbidden = new errors.Forbidden();
    expect(Forbidden).toMatchSnapshot();
    expect(Forbidden.code).toBe(403);
    expect(Forbidden.message).toBe('Forbidden');
  });
  it('allow Not Found error', () => {
    const NotFound = new errors.NotFound();
    expect(NotFound).toMatchSnapshot();
    expect(NotFound.code).toBe(404);
    expect(NotFound.message).toBe('Not Found');
  });
  it('allow Conflict error', () => {
    const Conflict = new errors.Conflict();
    expect(Conflict).toMatchSnapshot();
    expect(Conflict.code).toBe(409);
    expect(Conflict.message).toBe('Conflict');
  });
  it('allow Unsupported Media Type error', () => {
    const UnsupportedMediaType = new errors.UnsupportedMediaType();
    expect(UnsupportedMediaType).toMatchSnapshot();
    expect(UnsupportedMediaType.code).toBe(415);
    expect(UnsupportedMediaType.message).toBe('Unsupported Media Type');
  });
});
