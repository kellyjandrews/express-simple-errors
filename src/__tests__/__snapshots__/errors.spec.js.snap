exports[`Error Classes allow Bad Request error 1`] = `[Error: Bad Request]`;

exports[`Error Classes allow Conflict error 1`] = `[Error: Conflict]`;

exports[`Error Classes allow Forbidden error 1`] = `[Error: Forbidden]`;

exports[`Error Classes allow Not Found error 1`] = `[Error: Not Found]`;

exports[`Error Classes allow Unauthorized error 1`] = `[Error: Unauthorized]`;

exports[`Error Classes allow Unsupported Media Type error 1`] = `[Error: Unsupported Media Type]`;
